File: src/main/java/org/hwabeag/playerfly/schedules/PlayerFlyTask.java
package org.hwabeag.playerfly.schedules;

import java.util.Objects;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Player;
import org.hwabeag.playerfly.config.ConfigManager;

public class PlayerFlyTask implements Runnable {
    FileConfiguration PlayerFlyConfig = ConfigManager.getConfig("playerfly");
    String Prefix = ChatColor.translateAlternateColorCodes('&', Objects.requireNonNull(PlayerFlyConfig.getString("playerfly.prefix")));

    public void run() {
        for (Player player : Bukkit.getOnlinePlayers()) {
            String name = player.getName();
            if (Objects.equals(PlayerFlyConfig.getString("플라이." + name + ".관리"), "온")) {
                if (PlayerFlyConfig.getBoolean("playerfly.fly-infinity")) {
                    player.sendActionBar(Prefix + ChatColor.translateAlternateColorCodes('&',"&7당신은 플라이를 이용중입니다."));
                } else {
                    if (PlayerFlyConfig.getInt("플라이." + name + ".남은시간") > 0) {
                        int TimeNumber = PlayerFlyConfig.getInt("플라이." + name + ".남은시간");
                        PlayerFlyConfig.set("플라이." + name + ".남은시간", TimeNumber - 1);
                        ConfigManager.saveConfigs();
                        player.sendActionBar(Prefix + ChatColor.translateAlternateColorCodes('&',"&7남은 플라이: &a" + TimeNumber + " &7분"));
                    } else {
                        PlayerFlyConfig.set("플라이." + name + ".남은시간", 0);
                        PlayerFlyConfig.set("플라이." + name + ".관리", "오프");
                        ConfigManager.saveConfigs();
                        player.setFlying(false);
                        player.setAllowFlight(false);
                        String s = ChatColor.translateAlternateColorCodes('&', "&a&l[플라이]");
                        String s1 = ChatColor.translateAlternateColorCodes('&', "&a&l- &7당신은 플라이를 모두 소모하였습니다.");
                        player.sendTitle(s, s1);
                    }
                }
            }
        }
    }
}

File: src/main/kotlin/org/hwabeag/playerfly/schedules/PlayerFlyTask.kt
package org.hwabeag.playerfly.schedules

import org.bukkit.Bukkit
import org.bukkit.scheduler.BukkitRunnable
import org.hwabeag.playerfly.PlayerFlyPlugin

class PlayerFlyTask(private val plugin: PlayerFlyPlugin) : BukkitRunnable() {
    override fun run() {
        for (player in Bukkit.getOnlinePlayers()) {
            val data = plugin.databaseManager.getData(player.uniqueId, player.name)
            if (!data.enabled) {
                continue
            }

            if (!plugin.isManagedFlight(player.uniqueId)) {
                plugin.markManagedFlight(player.uniqueId)
            }

            if (plugin.config.getBoolean("playerfly.fly-infinity")) {
                player.sendActionBar("${plugin.prefix()} 무제한 플라이 사용 중입니다.")
                continue
            }

            if (data.minutes > 0) {
                val next = data.minutes - 1
                plugin.databaseManager.setMinutes(player.uniqueId, player.name, next)
                player.sendActionBar("${plugin.prefix()} 남은 플라이 시간: ${next}분")
                continue
            }

            plugin.databaseManager.setMinutes(player.uniqueId, player.name, 0)
            plugin.databaseManager.setEnabled(player.uniqueId, player.name, false)
            if (plugin.isManagedFlight(player.uniqueId)) {
                player.isFlying = false
                player.allowFlight = false
            }
            plugin.unmarkManagedFlight(player.uniqueId)
            player.sendTitle(plugin.color("&a&l[플라이]"), plugin.color("&a&l- 플라이 시간이 모두 소진되었습니다."))
        }
    }
}
